kotlin/filter-by-definition-expression-or-display-filter/src/main/java/com/esri/arcgisruntime/sample/filterbydefinitionexpressionordisplayfilter/MainActivity.kt

```diff
@@ -0,0 +1,178 @@
+
+import android.os.Bundle
+import android.util.Log
+import android.widget.Button
+import android.widget.TextView
+import android.widget.Toast
+import androidx.appcompat.app.AppCompatActivity
+import com.esri.arcgisruntime.ArcGISRuntimeEnvironment
+import com.esri.arcgisruntime.data.QueryParameters
+import com.esri.arcgisruntime.data.ServiceFeatureTable
+import com.esri.arcgisruntime.geometry.Point
+import com.esri.arcgisruntime.geometry.SpatialReferences
+import com.esri.arcgisruntime.layers.DisplayFilter
+import com.esri.arcgisruntime.layers.FeatureLayer
+import com.esri.arcgisruntime.layers.ManualDisplayFilterDefinition
+import com.esri.arcgisruntime.loadable.LoadStatus
+import com.esri.arcgisruntime.mapping.ArcGISMap
+import com.esri.arcgisruntime.mapping.BasemapStyle
+import com.esri.arcgisruntime.mapping.Viewpoint
+import com.esri.arcgisruntime.mapping.view.MapView
+import com.esri.arcgisruntime.sample.filterbydefinitionexpressionordisplayfilter.databinding.ActivityMainBinding
+
+
+class MainActivity : AppCompatActivity() {
+
+    private val TAG = MainActivity::class.java.simpleName
+
+    private val activityMainBinding by lazy {
+        ActivityMainBinding.inflate(layoutInflater)
+    }
+
+    private val featureCountTV: TextView by lazy {
+        activityMainBinding.featureCountTV
+    }
+
+    private val expressionButton: Button by lazy {
+        activityMainBinding.expressionButton
+    }
+    private val filterButton: Button by lazy {
+        activityMainBinding.filterButton
+    }
+
+    private val resetButton: Button by lazy {
+        activityMainBinding.resetButton
+    }
+
+    private val mapView: MapView by lazy {
+        activityMainBinding.mapView
+    }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(activityMainBinding.root)
+
+        // authentication with an API key or named user is required to access basemaps and other
+        // location services
+        ArcGISRuntimeEnvironment.setApiKey(BuildConfig.API_KEY)
+
+        // set the FeatureLayer using FeatureTable created from the URL to an ArcGIS feature service
+        val featureLayer = FeatureLayer(ServiceFeatureTable("https://services2.arcgis.com/ZQgQTuoyBrtmoGdP/arcgis/rest/services/SF_311_Incidents/FeatureServer/0"))
+
+        // set the map to be displayed in the layout's MapView
+        // with the BasemapType topographic
+        val topographicMap = ArcGISMap(BasemapStyle.ARCGIS_TOPOGRAPHIC).apply {
+            //add a feature layer to the map
+            operationalLayers.add(featureLayer)
+            addDoneLoadingListener {
+                if (loadStatus != LoadStatus.LOADED) {
+                    Log.e(TAG, "Failed to load the feature layer map")
+                    Toast.makeText(this@MainActivity, "Failed to load the feature layer map", Toast.LENGTH_SHORT).show()
+                }
```

Comment:
This is checking if the map loaded, not the feature layer. I think this is a confusing check here and can really just be removed

Reply:
Removed!